sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClient.java

```diff
@@ -32,9 +43,16 @@
  * HttpClient implementation for OkHttp.
  */
 class OkHttpAsyncHttpClient implements HttpClient {
-    final OkHttpClient httpClient;
-    //
+
     private static final Mono<okio.ByteString> EMPTY_BYTE_STRING_MONO = Mono.just(okio.ByteString.EMPTY);
+    /**
+     * This constant defines a size of Flux based request where buffering in memory becomes less performant
+     * than streaming (which involves thread hops). The value has been established experimentally using
+     * Storage benchmarks.
+     */
+    private static final long BUFFERED_FLUX_REQUEST_THRESHOLD = 100 * 1024L;
```

Comment:
Will this buffer size perform better than streaming in all environments? I am curious if the thread hops are the same across different platforms and devices. Also, will the perf be significantly better with larger buffer in some environments and if so, should this become a builder config to allow customers to set this?